hieroplatform/src/test/java/org/hiero/sketch/HistogramTest.java

```diff
@@ -0,0 +1,188 @@
+        size = 0;
+        size1 = 0;
+        size2 =0;
+        for (int i = 0; i < bucketnum; i++)
+            for (int j = 0; j< bucketnum; j++)
+                size += hist2.getBucket(i,j).getCount();
+        size += hist2.getMissingData();
+        size += hist2.getOutOfRange();
+        for (int i = 0; i < hist2.getNumOfBucketsD1(); i++)
+            size1 += hist2.getMissingHistogramD1().getBucket(i).getCount();
+        size1 += hist2.getMissingHistogramD1().getOutOfRange();
+        for (int i = 0; i < hist2.getNumOfBucketsD2(); i++)
+            size2 += hist2.getMissingHistogramD2().getBucket(i).getCount();
+        size2 += hist2.getMissingHistogramD2().getOutOfRange();
+
+        assertEquals(size + size1 + size2, 30000);
+        Histogram2DHeavy hist3 = new Histogram2DHeavy(buckDes1, buckDes2);
+        DoubleArrayColumn col5 = DoubleArrayTest.generateDoubleArray(20000);
+        DoubleArrayColumn col6 = DoubleArrayTest.generateDoubleArray(20000);
+        FullMembership fmap2 = new FullMembership(20000);
+        hist3.createSampleHistogram(col5, col6, null, null, fmap2, 0.1);
+    }
+
+    @Test
+    public void testHeatMap() {
+        final int bucketnum = 110;
+        BucketsDescriptionEqSize buckDes1 = new BucketsDescriptionEqSize(0,100,bucketnum);
+        BucketsDescriptionEqSize buckDes2 = new BucketsDescriptionEqSize(0,100,bucketnum);
+        HeatMap hm = new HeatMap(buckDes1, buckDes2);
+        DoubleArrayColumn col1 = this.generateDoubleArray(10000, 5);
+        DoubleArrayColumn col2 = this.generateDoubleArray(10000, 3);
+        FullMembership fmap = new FullMembership(10000);
+        hm.createHistogram(col1, col2, null, null, fmap);
+        int size0 = 0;
+        int size1 = 0;
+        int size2 = 0;
+        for (int i = 0; i < hm.getNumOfBucketsD1(); i++)
+            for (int j = 0; j < hm.getNumOfBucketsD2(); j++)
+                size0 += hm.getBucket(i,j);
+        size0 += hm.getMissingData();
+        size0 += hm.getOutOfRange();
+        for (int i = 0; i < hm.getNumOfBucketsD1(); i++)
+            size1 += hm.getMissingHistogramD1().getBucket(i);
+        size1 += hm.getMissingHistogramD1().getOutOfRange();
+        for (int i = 0; i < hm.getNumOfBucketsD2(); i++)
+            size2 += hm.getMissingHistogramD2().getBucket(i);
+        size2 += hm.getMissingHistogramD2().getOutOfRange();
+        assertEquals(10000, size0 + size1 + size2);
+
+        HeatMap hm1 = new HeatMap(buckDes1, buckDes2);
+        DoubleArrayColumn col3 = DoubleArrayTest.generateDoubleArray(20000);
+        DoubleArrayColumn col4 = DoubleArrayTest.generateDoubleArray(20000);
+        FullMembership fmap1 = new FullMembership(20000);
+        hm1.createSampleHistogram(col3, col4, null, null, fmap1, 0.1);
+        HeatMap hm2 = hm.union(hm1);
+        assertEquals(hm.getSize() + hm1.getSize(), hm2.getSize());
+    }
+
+    private final ColumnDescription desc = new ColumnDescription("SQRT", ContentsKind.Double, true);
```

Comment:
Can you also please add a test for a string column with a suitable converter?
The most interesting will be the one with bucket metadata.